finesse/hardware/temperature/dp9800.py

```diff
@@ -0,0 +1,219 @@
+        sysflag: string representation of the system flag bitmask
+    """
+    try:
+        data_ascii = data.decode("ascii")
+    except Exception as e:
+        raise DP9800Error(e)
+
+    vals_begin = 3
+    vals_end = 74
+    etx_index = data.find(b"\x03")
+
+    vals = [Decimal(val) for val in data_ascii[vals_begin:vals_end].split()]
+
+    sysflag = bin(int(data_ascii[vals_end:etx_index], 16))
+
+    return (vals[1:], sysflag[2:])
+
+
+class DP9800Error(Exception):
+    """Indicates that an error occurred while communicating with the device."""
+
+
+class DP9800:
+    """An interface for DP9800 temperature readers.
+
+    The manual for this device is available at:
+    https://assets.omega.com/manuals/M5210.pdf
+    """
+
+    def __init__(self, serial: Serial) -> None:
+        """Create a new DP9800 from an existing serial device.
+
+        Args:
+            serial: Serial device
+        """
+        self.serial = serial
+
+        logging.info(f"Opened connection to DP9800 on port {self.serial.port}")
+        pub.sendMessage("temperature_monitor.open")
+        pub.subscribe(self.send_temperatures, "temperature_monitor.data.request")
+
+    def close(self) -> None:
+        """Close the connection to the device.
+
+        Raises:
+            SerialException: Error communicating with device
+        """
+        try:
+            self.serial.close()
+            pub.sendMessage("temperature_monitor.close")
+            logging.info("Closed connection to DP9800")
+        except SerialException as e:
+            raise DP9800Error(e)
+
+    def print_sysflag(self, sysflag: str) -> None:
+        """Print the settings of the device as stored in the system flag.
+
+        The system flag is stored as a bit mask with the format TxxLxSAF,
+        where:
+            F - bit 0: temperature unit: 0 = deg C, 1 = deg F
+            A - bit 1: audible button:   0 = silence, 1 = audible
+            S - bit 2: autoscan:         0 = no scan, 1 = autoscan active
+            x - bit 3: must be 0
+            L - bit 4: logging:          0 = no logging, 1 = logging active
+            x - bit 5: must be 0
+            x - bit 6: must be 0
+            T - bit 7: instrument type:  0 = TC, 1 = PT
+
+        Args:
+            sysflag: string representation of the system flag bitmask
+        """
+        if sysflag != "":
+            instr_type = ["TC", "PT"][int(sysflag[0])]
+            logging_state = ["no logging", "logging active"][int(sysflag[3])]
+            scanning_state = ["no scan", "autoscan active"][int(sysflag[5])]
+            audible_state = ["silence", "audible"][int(sysflag[6])]
+            temp_unit = ["deg C", "deg F"][int(sysflag[7])]
+            print(f"Instrument type: {instr_type}")
+            print(f"Logging: {logging_state}")
+            print(f"Autoscan: {scanning_state}")
+            print(f"Audible button: {audible_state}")
+            print(f"Temperature unit: {temp_unit}")
```

Comment:
It's up to the user to provide sane input so you don't need to check it:

```suggestion
        instr_type = ["TC", "PT"][int(sysflag[0])]
        logging_state = ["no logging", "logging active"][int(sysflag[3])]
        scanning_state = ["no scan", "autoscan active"][int(sysflag[5])]
        audible_state = ["silence", "audible"][int(sysflag[6])]
        temp_unit = ["deg C", "deg F"][int(sysflag[7])]
        print(f"Instrument type: {instr_type}")
        print(f"Logging: {logging_state}")
        print(f"Autoscan: {scanning_state}")
        print(f"Audible button: {audible_state}")
        print(f"Temperature unit: {temp_unit}")
```